core/chains/evm/logpoller/log_poller.go

```diff
@@ -50,6 +51,14 @@ type LogPoller interface {
 	LogsDataWordGreaterThan(eventSig common.Hash, address common.Address, wordIndex int, wordValueMin common.Hash, confs int, qopts ...pg.QOpt) ([]Log, error)
 }
 
+type LogPollerTest interface {
```

Comment:
I don't love this but I think its the best compromise to permit synchronous whitebox testing wherever we want. Upside is we can avoid Eventually + the inevitable flakey tests and its still relatively restricted (we expect all clients to use only the LogPoller intf).

Reply:
Why not use casting to concrete types in the whitebox tests instead?

Reply:
What's the downside of Eventually? I was excited when I found it recently, as it seemed like the perfect tool for fixing a flaky test, enabling me to avoid using sleeps.  I assumed it was something I should use more often, but I guess not?

Reply:
@samsondav I should've said synchronous white or blackbox testing - exposing it helps for both

Reply:
@reductionista eventually is the best thing to use for the handful of places where we have to test some runloop/channel logic, but we still want to minimize its use in general because its slow (especially trying to debug a failure and you have to wait for some 30s/minute timeout each run) and you still have to set a timeout which can flake (we've had to bump timeouts before)